packages/bitcore-node/src/models/coin.ts

```diff
@@ -90,13 +88,31 @@ class CoinModel extends BaseModel<ICoin> {
         }
       ])
       .toArray();
-    return result.reduce((acc, cur) => { 
-      acc[cur._id] = cur.balance; 
+    return result.reduce((acc, cur) => {
+      acc[cur._id] = cur.balance;
       acc.balance += cur.balance;
-      return acc; 
+      return acc;
     }, { confirmed: 0, unconfirmed: 0, balance: 0 }) as { confirmed: number, unconfirmed: number, balance: number };
   }
 
+  async getBalanceAtTime(params: { query: any, time: string, chain: string, network: string }) {
+    let { query, time, chain, network } = params;
+    const block = await BlockStorage.collection.findOne({
+      $query: {
+        chain,
+        network,
+        time: { $lte: new Date(time) }
+      },
+      $orderBy: { _id: -1 }
+    });
+    const blockHeight = block!.height
+    const combinedQuery = Object.assign({}, {
+      $or: [{ spentHeight: { $gt: blockHeight } }, { spentHeight: { $lt: SpentHeightIndicators.minimum } }],
+      mintHeight: { $lte: blockHeight }
+    }, query);
+    return this.getBalance({ query: combinedQuery });
```

Comment:
A good unit test would be to stub the aggregation, and then expect the query passed into to aggregate to match a certain object for each method.